bc_obps/compliance/service/compliance_report_version_service.py

```diff
@@ -151,3 +156,32 @@ def get_operation_by_compliance_report_version(compliance_report_version_id: int
     @staticmethod
     def get_obligation_by_compliance_report_version(compliance_report_version_id: int) -> ComplianceObligation:
         return ComplianceObligation.objects.get(compliance_report_version__id=compliance_report_version_id)
+
+    @classmethod
+    def get_compliance_report_versions_for_previously_owned_operations(
+        cls, user_guid: UUID
+    ) -> QuerySet[ComplianceReportVersion]:
+        """
+        Fetches all compliance report versions that a user should have access to based on historical operator -> operation ownership relationships
+        """
+
+        all_historical_report_versions = ComplianceReportVersion.objects.none()
+        operator_id = UserOperator.objects.get(user_id=user_guid).operator_id
+        previously_owned_operations = (
+            OperationDesignatedOperatorTimelineDataAccessService.get_previously_owned_operations_by_operator(
+                operator_id=operator_id
+            )
+        )
+
+        for operation_ownership in previously_owned_operations:
+            historical_report_versions = ComplianceReportVersion.objects.filter(
+                compliance_report__report__operation_id=operation_ownership.operation_id,
+                compliance_report__compliance_period__end_date__gte=operation_ownership.start_date,
+                compliance_report__compliance_period__end_date__lt=operation_ownership.end_date,
+            ).select_related(
+                'compliance_report__report__operation',
+                'compliance_report__compliance_period',
+                'compliance_earned_credit',
+            )
+            all_historical_report_versions = all_historical_report_versions.union(historical_report_versions)
+        return all_historical_report_versions
```

Comment:
Suggestion: This approach is slightly more optimized, as it avoids defining an empty object first and then performing a union later.

```python
    query = Q()
    for operation_ownership in previously_owned_operations:
        query |= (
            Q(compliance_report__report__operation_id=operation_ownership.operation_id) &
            Q(compliance_report__compliance_period__start_date__lte=operation_ownership.end_date) &
            Q(compliance_report__compliance_period__end_date__gte=operation_ownership.start_date)
        )
    
    return ComplianceReportVersion.objects.filter(query).select_related(
        'compliance_report__report__operation',
        'compliance_report__compliance_period',
        'compliance_earned_credit',
    )
```